### ISSUE TITLE
VS2019 Remote debugging: Attach fails on linux-arm with .NET 5.0.2 (Error 0x80131c4f)

### ISSUE THREAD

**Issue by XXXBold** [dotnet/runtime#47848]
### Description
I'm not able to attach a debugger to a remote process on a Toradex T30 with linux-arm. I'm using a simple testproject (see below) to test this. Running the .dll with `dotnet Test_net5.0.dll` from the CLI on the device does work.

I compile & publish the Project on my Windows 10 machine. Following settings are used to publish:
![vs-publish-options](https://user-images.githubusercontent.com/30773753/106901451-0f53ba00-66f8-11eb-904c-3ccdff50a087.PNG)

After that, files are copied manually over SSH to /home/root/dn5test/ on the client, using pscp.

For launching, I'm using a launch.json file and the command:
`DebugAdapterHost.Launch /LaunchJson:launch.json /ConfigurationName:RL`

The process seems to start on the target as desired (verified using htop):
![htop](https://user-images.githubusercontent.com/30773753/106899088-3ceb3400-66f5-11eb-91c9-0f8e29c70c94.PNG)

But on Visual Studio, I always get the Error mentioned above, when starting or also when I try to attach to the running process while it's running:
![vs-error](https://user-images.githubusercontent.com/30773753/106901850-7d987c80-66f8-11eb-8782-6d8e9da9edfa.PNG)

When I stop debugging in VS, the process gets automatically terminated on the client aswell.

I already opened an issue for this before (with an earlier version of .NET 5), which was closed then: https://github.com/dotnet/runtime/issues/44941#issue-746583251 and I cannot open it anymore.

Tell me if you need more information, and I'll provide them if I can.

#### Steps to reproduce
- Install .NET shared framework on linux client
- Publish compiled VB-Example Project to client
- Run command for remote debugging
- Here you go, for me it fails 100% of the time

### Configuration
Host (Using to develop & debug):
Windows 10 Pro 2004
Visual Studio 2019 V16.8.4

Client (Used as target to run .NET applications):
Toradex T30 on IRIS Carrier board: https://developer.toradex.com/software/linux/linux-software#tab-bsp-28
Running custom Yocto Linux based on Ångström v2017.12 (BSP 2.8v7, Yocto version 2.4): https://developer.toradex.com/knowledge-base/board-support-package/openembedded-core
.NET 5.0.2 runtime installed from https://dotnet.microsoft.com/download/dotnet/5.0 (linux-arm) at /usr/share/dotnet

### Regression?
Earlier versions don't work either (3.1, 5.0.1 tested).

#### Project source (Program.vb)
```
Imports System

Module Program
    Sub Main(args As String())
        While 1
            Console.WriteLine("Hello World!")
            Task.Delay(2000).Wait()
        End While
    End Sub
End Module
```

#### dotnet --info on client
```
Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.NETCore.App 5.0.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

#### launch.json-file
```
{
  "__comment": "Run in VS2019 terminal using DebugAdapterHost.Launch /LaunchJson:<path> /ConfigurationName:RL",
  "version": "0.2.0",
  "adapter": "plink.exe",
  "adapterArgs": "root@10.10.11.55 -batch -T /home/root/vsdbg/vsdbg --interpreter=vscode",
  "configurations": [
    {
      "name": "RL",
      "type": "coreclr",
      "request": "launch",
      "cwd": "~/dn5test",
      "program": "/usr/share/dotnet/dotnet",
      "args": "Test_net5.0.dll"
    },
    {
      "name": "RL_AT",
      "type": "coreclr",
      "request": "attach",
      "processId": "${command:pickRemoteProcess}",
      "pipeTransport": {
        "pipeProgram": "ssh",
        "pipeArgs": [
          "-T",
          "root@10.10.11.55"
        ],
        "debuggerPath": "/home/root/vsdbg/vsdbg",
        "pipeCwd": "${workspaceRoot}",
        "quoteArgs": true
      },
      "justMyCode": false,
      "symbolOptions": {
        "searchPaths": [ "~/dn5test" ],
        "searchMicrosoftSymbolServer": false
      },
      "requireExactSource": false
    }
  ]
}
```

#### Debug Adapter Host Log
```
1> DebugAdapterHost version: 16.8.40922.1 commit:6ac6030cbe2c515a73b952feec2d243d418d0915
 1> State: Unset => Initializing
 1> Sending VS event: IDebugEngineCreateEvent2 (fe5b734c-759d-4e59-ab04-f103343bdd06)
 1> Sending VS event: IDebugProgramCreateEvent2 (96cd11ee-ecd4-4e89-957e-b5d496fc4139)
 1> Starting 'plink.exe' with arguments 'root@10.10.11.55 -batch -T /home/root/vsdbg/vsdbg --interpreter=vscode'
 1> WARNING: 'Language' not specified in engine registration!
 1> WARNING: 'LanguageId' not specified in engine registration!
 1> [DebugAdapter] --> C (initialize-1): {"type":"request","command":"initialize","arguments":{"pathFormat":"path","clientID":"visualstudio","clientName":"Visual Studio","adapterID":"coreclr","locale":"en-US","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsRunInTerminalRequest":true,"supportsMemoryReferences":true,"supportsProgressReporting":true,"SupportsMessageBox":true,"supportsHandshakeRequest":true,"supportsVsAdditionalBreakpointBinds":true,"supportsHitCountsChange":true,"supportsVsCustomMessages":true},"seq":1}
 1> WARNING: Request 'initialize-1' has not received a response within 1000 ms!
 1> [DebugAdapter] <--   R (initialize-1) [1693 ms]: {"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsHitConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"exceptionBreakpointFilters":[{"filter":"all","label":"All Exceptions","default":false},{"filter":"user-unhandled","label":"User-Unhandled Exceptions","default":true}],"supportsSetVariable":true,"supportsGotoTargetsRequest":true,"supportsModulesRequest":true,"additionalModuleColumns":[{"attributeName":"vsLoadAddress","label":"Load Address","type":"string"},{"attributeName":"vsPreferredLoadAddress","label":"Preferred Load Address","type":"string"},{"attributeName":"vsModuleSize","label":"Module Size","type":"number"},{"attributeName":"vsLoadOrder","label":"Order","type":"number"},{"attributeName":"vsTimestampUTC","label":"Timestamp","type":"unixTimestampUTC"},{"attributeName":"vsIs64Bit","label":"64-bit","type":"boolean"},{"attributeName":"vsAppDomain","label":"AppDomain","type":"string"},{"attributeName":"vsAppDomainId","label":"AppDomainId","type":"number"}],"supportedChecksumAlgorithms":["MD5","SHA1","SHA256"],"supportsExceptionOptions":true,"supportsValueFormattingOptions":true,"supportsExceptionInfoRequest":true,"supportTerminateDebuggee":true,"supportsSetExpression":true,"supportsReadMemoryRequest":true,"supportsCancelRequest":true,"supportsExceptionConditions":true,"supportsLoadSymbolsRequest":true,"supportsModuleSymbolSearchLog":true,"supportsDebuggerProperties":true,"supportsSetSymbolOptions":true,"supportsHitBreakpointIds":true,"supportsVsIndividualBreakpointOperations":true,"supportsVsBreakpointLanguage":true,"supportsSetHitCount":true,"supportsVsCustomMessages":true,"supportsEvaluationOptions":true,"supportsExceptionStackTrace":true,"memoryReferencesAreAddresses":true,"supportsObjectFavorites":true,"supportsObjectId":true,"supportsVariableEnumerators":false},"seq":1}
 1> Sending VS event: IDebugEngineMetricsUpdatedEvent166 (45b79d58-0620-410f-8f39-c7c8c155e36c)
 1> [DebugAdapter] --> C (launch-2): {"type":"request","command":"launch","arguments":{"name":"RL","type":"coreclr","request":"launch","cwd":"~/dn5test","program":"/usr/share/dotnet/dotnet","args":"Test_net5.0.dll"},"seq":2}
 1> [DebugAdapter] <--   C (handshake-2): {"type":"request","command":"handshake","arguments":{"value":"i6YgC8zNJPztphbcK0TNhiFsh+aFnYU0b0ZHS1ytWEc="},"seq":2}
 1> [DebugAdapter] --> R (handshake-2): {"type":"response","request_seq":2,"success":true,"command":"handshake","body":{"signature":"5900T2r0WtT4ec1sw0EDw0nvLpmvTN8zUmfmJq0zPRV+fY="},"seq":3}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-------------------------------------------------------------------\nYou may only use the Microsoft .NET Core Debugger (vsdbg) with\nVisual Studio Code, Visual Studio or Visual Studio for Mac software\nto help you develop and test your applications.\n-------------------------------------------------------------------\n","severity":"ok"},"seq":3}
 1> Sending VS event: IDebugMessageEvent2 (3bdb28cf-dbd2-4d24-af03-01072b67eb9e)
 1> [DebugAdapter] <--   R (launch-2) [14 ms]: {"type":"response","request_seq":2,"success":true,"command":"launch","seq":4}
 1> State: Initializing => Running
 1> [DebugAdapter] <--   E (initialized): {"type":"event","event":"initialized","body":{},"seq":5}
 1> [DebugAdapter] --> C (setDebuggerProperty-4): {"type":"request","command":"setDebuggerProperty","arguments":{"AllowOutOfProcessSymbols":1},"seq":4}
 1> [DebugAdapter] --> C (setDebuggerProperty-5): {"type":"request","command":"setDebuggerProperty","arguments":{"DisableJITOptimization":0},"seq":5}
 1> [DebugAdapter] --> C (setDebuggerProperty-6): {"type":"request","command":"setDebuggerProperty","arguments":{"EnableFastEvaluate":1},"seq":6}
 1> [DebugAdapter] --> C (setDebuggerProperty-7): {"type":"request","command":"setDebuggerProperty","arguments":{"InterpreterOptions":1},"seq":7}
 1> [DebugAdapter] --> C (setDebuggerProperty-8): {"type":"request","command":"setDebuggerProperty","arguments":{"JustMyCodeStepping":1},"seq":8}
 1> [DebugAdapter] --> C (setDebuggerProperty-9): {"type":"request","command":"setDebuggerProperty","arguments":{"StopOnExceptionCrossingManagedBoundary":0},"seq":9}
 1> [DebugAdapter] --> C (setDebuggerProperty-10): {"type":"request","command":"setDebuggerProperty","arguments":{"WarnIfNoUserCodeOnLaunch":1},"seq":10}
 1> [DebugAdapter] --> C (setDebuggerProperty-11): {"type":"request","command":"setDebuggerProperty","arguments":{"EnableStepFiltering":true},"seq":11}
 1> [DebugAdapter] --> C (setSymbolOptions-12): {"type":"request","command":"setSymbolOptions","arguments":{"symbolOptions":{"cachePath":"","moduleFilter":{"mode":"loadAllButExcluded","excludedModules":[""]}}},"seq":12}
 1> [DebugAdapter] <--   R (setDebuggerProperty-4) [9 ms]: {"type":"response","request_seq":4,"success":true,"command":"setDebuggerProperty","body":{},"seq":6}
 1> [DebugAdapter] --> C (setFunctionBreakpoints-13): {"type":"request","command":"setFunctionBreakpoints","arguments":{"breakpoints":[]},"seq":13}
 1> [DebugAdapter] <--   R (setDebuggerProperty-5) [11 ms]: {"type":"response","request_seq":5,"success":true,"command":"setDebuggerProperty","body":{},"seq":7}
 1> Sending VS event: IDebugMessagableProgramReadyEvent167 (68bdc978-c107-46e1-80ad-29eb7af9472d)
 1> [DebugAdapter] <--   R (setDebuggerProperty-6) [11 ms]: {"type":"response","request_seq":6,"success":true,"command":"setDebuggerProperty","body":{},"seq":8}
 1> [DebugAdapter] <--   R (setDebuggerProperty-7) [11 ms]: {"type":"response","request_seq":7,"success":true,"command":"setDebuggerProperty","body":{},"seq":9}
 1> [DebugAdapter] <--   R (setDebuggerProperty-8) [11 ms]: {"type":"response","request_seq":8,"success":true,"command":"setDebuggerProperty","body":{},"seq":10}
 1> [DebugAdapter] <--   R (setDebuggerProperty-9) [11 ms]: {"type":"response","request_seq":9,"success":true,"command":"setDebuggerProperty","body":{},"seq":11}
 1> [DebugAdapter] <--   R (setDebuggerProperty-10) [11 ms]: {"type":"response","request_seq":10,"success":true,"command":"setDebuggerProperty","body":{},"seq":12}
 1> [DebugAdapter] <--   R (setDebuggerProperty-11) [11 ms]: {"type":"response","request_seq":11,"success":true,"command":"setDebuggerProperty","body":{},"seq":13}
 1> [DebugAdapter] <--   R (setSymbolOptions-12) [153 ms]: {"type":"response","request_seq":12,"success":true,"command":"setSymbolOptions","seq":14}
 1> [DebugAdapter] <--   R (setFunctionBreakpoints-13) [153 ms]: {"type":"response","request_seq":13,"success":true,"command":"setFunctionBreakpoints","body":{"breakpoints":[]},"seq":15}
 1> [DebugAdapter] --> C (setExceptionBreakpoints-14): {"type":"request","command":"setExceptionBreakpoints","arguments":{"filters":[],"exceptionOptions":[]},"seq":14}
 1> [DebugAdapter] <--   R (setExceptionBreakpoints-14) [12 ms]: {"type":"response","request_seq":14,"success":true,"command":"setExceptionBreakpoints","seq":16}
 1> [DebugAdapter] --> C (configurationDone-15): {"type":"request","command":"configurationDone","arguments":{},"seq":15}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/ProcessCreate","data":{"VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"e05d318","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.Version":"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3"}},"seq":17}
 1> [DebugAdapter] <--   E (telemetryDetails): {"type":"event","event":"telemetryDetails","body":{"telemetryEventName":"vs/diagnostics/debugger/process/context","telemetryProperties":{"VS.Diagnostics.Debugger.AD7Process.Architecture":"ARM"}},"seq":18}
 1> Sending VS event: IDebugTelemetryDetailsEvent150 (8f2652b2-cd3c-4aed-a946-a3db6f379412)
 1> [DebugAdapter] <--   E (process): {"type":"event","event":"process","body":{"name":"/usr/share/dotnet/dotnet","systemProcessId":1471,"startMethod":"launch","pointerSize":32},"seq":19}
 1> Sending VS event: IDebugProcessInfoUpdatedEvent158 (96c242fc-f584-4c3e-8fed-384d3d13ef36)
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/Launch","data":{"VS.Diagnostics.Debugger.vsdbg.AdapterId":"coreclr","VS.Diagnostics.Debugger.vsdbg.Launch.Duration":220,"VS.Diagnostics.Debugger.vsdbg.VisualizerFileUsed":false,"VS.Diagnostics.Debugger.vsdbg.TargetType":"Live","VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"e05d318","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.SourceFileMappings":0,"VS.Diagnostics.Debugger.vsdbg.Version":"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3"}},"seq":20}
 1> [DebugAdapter] <--   R (configurationDone-15) [32 ms]: {"type":"response","request_seq":15,"success":true,"command":"configurationDone","seq":21}
 1> Sending VS event: IDebugLoadCompleteEvent2 (b1844850-1349-45d4-9f12-495212f5eb0b)
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"messageBox","output":"Unable to attach to CoreCLR. Unknown Error: 0x80131c4f","severity":"error"},"seq":22}
 1> Sending VS event: IDebugMessageEvent2 (3bdb28cf-dbd2-4d24-af03-01072b67eb9e)
```

**Comment by hoyosjs** [comment]
The Toradex T30 is a 1gb memory, right? If so, this is a new issue.

However, I've been trying to repro for a good time with VS and VS Code, and 3.1 and 5.0 on a couple of raspberry pi, which is all I have to test linux-arm. However, all works as intended and I can attach. They also have 1 GB of ram, but they are debian based (10). It's hard for me to say what's going on, it might just be something from the board/OS. Since those are not OS's that we support, a few things could be going on that we don't test in said distro. Not sure what we can do here other than get a debugger attached or do some logging. 

![image](https://user-images.githubusercontent.com/19413848/107002727-857c0f00-6740-11eb-9811-05d81c913cf0.png)

**Comment by XXXBold** [comment]
> 
> 
> The Toradex T30 is a 1gb memory, right?

That's correct, yes.


> 
> 
> It's hard for me to say what's going on, it might just be something from the board/OS. Since those are not OS's that we support, a few things could be going on that we don't test in said distro. Not sure what we can do here other than get a debugger attached or do some logging.

I understand that you cannot officially support any distro out there. However, if I can assist you with more information / logging, let me know what you need, I'm glad to help if possible.

**Comment by hoyosjs** [comment]
Do you have any native debuggers available on the machine?

**Comment by XXXBold** [comment]
There was no debugger installed on the image by default, but I just installed gdb (8.0.0) using the integrated opkg. Is this sufficent to proceed?

**Comment by hoyosjs** [comment]
Sure. Use you little infinite loop program. It probably will be easier in 3.1 as the logic was a lot easier. You'll need to modify the launch file a little to look like this, just replace paths, the IP/user + the key file, I think you use PLINK so that too, and also the path to vsdbg:

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "$adapter": "C:/Windows/SysNative/OpenSSH/ssh.exe",
      "$adapterArgs": "-T diag-pi3 /home/pi/vsdbg/vsdbg --pauseEngineForDebugger --interpreter=vscode",
      "name": "Launch REMOTE",
      "type": "coreclr",
      "request": "launch",
      "program": "dotnet",
      "args": [ "/home/pi/dn5test/ConsoleApp9.dll" ],
      "cwd": "/home/pi/dn5test/",
      "stopAtEntry": true,
      "console": "internalConsole",
      "justMyCode": false,
      "logging": {
        "engineLogging": true,
        "programOutput": true,
        "exceptions": true
      }
    }
  ]
}
```

1. download symbols for the runtime. Given that the arm device is problematic, a possibly easy thing to do is publish it self contained, as opposed to framework dependent. Then install [dotnet-symbol](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-symbol) on your Windows machine, and then from the publish directory run `dotnet symbol libmscordaccore.so -o .` and `dotnet symbol libmscordbi.so -o .` (there is a little problem with this right now). Then coppy all this to the arm device.
2. Clone https://github.com/dotnet/coreclr/ and check out the tag for the release you are debugging. (if debugging 3.1)
2. Start the debug session in VS. It will look stuck, that's OK.
2. attach GDB to that process (vsdbg) in the arm device.
4. Set a breakpoint at [`ClrDataAccess::GetDacGlobals`](https://github.com/dotnet/coreclr/blob/release/3.1/src/debug/daccess/daccess.cpp) and continue. 
5. The breakpoint should hit. Step a bit and you'll see at what point we return `CORDBG_E_MISSING_DEBUGGER_EXPORTS`. All the point where we throw said exception are under that method. Once you get here we'll have a better idea of that going on.

**Comment by XXXBold** [comment]
Thanks for your quick answer and the instructions!

#### Some questions: 
- Why should I clone coreclr from github? I've downloaded v3.1.11 from here: https://dotnet.microsoft.com/download/dotnet-core/3.1 and installed it on the device. Is that just needed to look at the source code or do I miss something?
- I'm not using public key auth on ssh, just entering the username (root) is enough to login (Device is inside local network only). Is that a problem?
- I've made some changes to your launch.json script (left out the --pauseEngineForDebugger for now)
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "$adapter": "plink.exe",
      "$adapterArgs": "root@10.10.11.55 -batch -T /home/root/vsdbg/vsdbg --interpreter=vscode",
      "name": "Launch REMOTE",
      "type": "coreclr",
      "request": "launch",
      "program": "dotnet",
      "args": [ "/home/root/dn31dbg/ConsoleApp1.dll" ],
      "cwd": "/home/root/dn31dbg/",
      "stopAtEntry": true,
      "console": "internalConsole",
      "justMyCode": false,
      "logging": {
        "engineLogging": true,
        "programOutput": true,
        "exceptions": true
      }
    }
  ]
}
```

Unfortunately I get an error when launching `DebugAdapterHost.Launch /LaunchJson:launch.json`:
#### Debug Adapter Host Log:
```
 1> DebugAdapterHost version: 16.8.40922.1 commit:6ac6030cbe2c515a73b952feec2d243d418d0915
 1> Starting 'plink.exe' with arguments 'root@10.10.11.55 -batch -T /home/root/vsdbg/vsdbg --interpreter=vscode'
 1> WARNING: 'Language' not specified in engine registration!
 1> WARNING: 'LanguageId' not specified in engine registration!
 1> [DebugAdapter] --> C (initialize-1): {"type":"request","command":"initialize","arguments":{"pathFormat":"path","clientID":"visualstudio","clientName":"Visual Studio","adapterID":"coreclr","locale":"en-US","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsRunInTerminalRequest":true,"supportsMemoryReferences":true,"supportsProgressReporting":true,"SupportsMessageBox":true,"supportsHandshakeRequest":true,"supportsVsAdditionalBreakpointBinds":true,"supportsHitCountsChange":true,"supportsVsCustomMessages":true},"seq":1}
 1> WARNING: Request 'initialize-1' has not received a response within 1000 ms!
 1> [DebugAdapter] <--   R (initialize-1) [1389 ms]: {"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsHitConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"exceptionBreakpointFilters":[{"filter":"all","label":"All Exceptions","default":false},{"filter":"user-unhandled","label":"User-Unhandled Exceptions","default":true}],"supportsSetVariable":true,"supportsGotoTargetsRequest":true,"supportsModulesRequest":true,"additionalModuleColumns":[{"attributeName":"vsLoadAddress","label":"Load Address","type":"string"},{"attributeName":"vsPreferredLoadAddress","label":"Preferred Load Address","type":"string"},{"attributeName":"vsModuleSize","label":"Module Size","type":"number"},{"attributeName":"vsLoadOrder","label":"Order","type":"number"},{"attributeName":"vsTimestampUTC","label":"Timestamp","type":"unixTimestampUTC"},{"attributeName":"vsIs64Bit","label":"64-bit","type":"boolean"},{"attributeName":"vsAppDomain","label":"AppDomain","type":"string"},{"attributeName":"vsAppDomainId","label":"AppDomainId","type":"number"}],"supportedChecksumAlgorithms":["MD5","SHA1","SHA256"],"supportsExceptionOptions":true,"supportsValueFormattingOptions":true,"supportsExceptionInfoRequest":true,"supportTerminateDebuggee":true,"supportsSetExpression":true,"supportsReadMemoryRequest":true,"supportsCancelRequest":true,"supportsExceptionConditions":true,"supportsLoadSymbolsRequest":true,"supportsModuleSymbolSearchLog":true,"supportsDebuggerProperties":true,"supportsSetSymbolOptions":true,"supportsHitBreakpointIds":true,"supportsVsIndividualBreakpointOperations":true,"supportsVsBreakpointLanguage":true,"supportsSetHitCount":true,"supportsVsCustomMessages":true,"supportsEvaluationOptions":true,"supportsExceptionStackTrace":true,"memoryReferencesAreAddresses":true,"supportsObjectFavorites":true,"supportsObjectId":true,"supportsVariableEnumerators":false},"seq":1}
 1> [DebugAdapter] --> C (launch-2): {"type":"request","command":"launch","arguments":{"name":"Launch REMOTE","type":"coreclr","request":"launch","program":"dotnet","args":["/home/root/dn31dbg/ConsoleApp1.dll"],"cwd":"/home/root/dn31dbg/","stopAtEntry":true,"console":"internalConsole","justMyCode":false,"logging":{"engineLogging":true,"programOutput":true,"exceptions":true}},"seq":2}
 1> [DebugAdapter] <--   C (handshake-2): {"type":"request","command":"handshake","arguments":{"value":"MOj/6FsTcEGyx5AYrSRDTrNQM0h1Rk8i4Xy0aCUrhWU="},"seq":2}
 1> [DebugAdapter] --> R (handshake-2): {"type":"response","request_seq":2,"success":true,"command":"handshake","body":{"signature":"93959a+gFuOZFsItzAQMPqUeD2aiS6LBd+ZFn5DARi2II8="},"seq":3}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":3,\"type\":\"event\",\"event\":\"output\",\"body\":{\"category\":\"console\",\"output\":\"-------------------------------------------------------------------\\nYou may only use the Microsoft .NET Core Debugger (vsdbg) with\\nVisual Studio Code, Visual Studio or Visual Studio for Mac software\\nto help you develop and test your applications.\\n-------------------------------------------------------------------\\n\",\"severity\":\"ok\"}}\n"},"seq":4}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-------------------------------------------------------------------\nYou may only use the Microsoft .NET Core Debugger (vsdbg) with\nVisual Studio Code, Visual Studio or Visual Studio for Mac software\nto help you develop and test your applications.\n-------------------------------------------------------------------\n","severity":"ok"},"seq":3}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":5,\"type\":\"event\",\"event\":\"output\",\"body\":{\"category\":\"telemetry\",\"output\":\"VS/Diagnostics/Debugger/vsdbg/LaunchFailed\",\"data\":{\"VS.Diagnostics.Debugger.vsdbg.AdapterId\":\"coreclr\",\"VS.Diagnostics.Debugger.vsdbg.ErrorCode\":1002,\"VS.Diagnostics.Debugger.vsdbg.Distribution.Version\":\"\",\"VS.Diagnostics.Debugger.vsdbg.Distribution.Name\":\"e05d318\",\"VS.Diagnostics.Debugger.vsdbg.OSFamily\":\"Linux\",\"VS.Diagnostics.Debugger.vsdbg.Version\":\"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3\"}}}\n"},"seq":6}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/LaunchFailed","data":{"VS.Diagnostics.Debugger.vsdbg.AdapterId":"coreclr","VS.Diagnostics.Debugger.vsdbg.ErrorCode":1002,"VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"e05d318","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.Version":"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3"}},"seq":5}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":7,\"type\":\"response\",\"request_seq\":2,\"success\":false,\"command\":\"launch\",\"message\":\"launch: program '' does not exist.\"}\n"},"seq":8}
 1> [DebugAdapter] <--   R (launch-2) [18 ms]: {"type":"response","request_seq":2,"success":false,"command":"launch","message":"launch: program '' does not exist.","seq":7}
 1> ERROR: Unexpected error

AggregateException: One or more errors occurred.

Aggregate exception: 
    DebugAdapterLaunchException: Failed to launch debug adapter.  Additional information may be available in the output window.

    Failure Location: LaunchRequest
    Inner Exception: 
        ProtocolException: launch: program '' does not exist.
Inner Exception: 
    DebugAdapterLaunchException: Failed to launch debug adapter.  Additional information may be available in the output window.

Microsoft.VisualStudio.Debugger.VSCodeDebuggerHost.Engine.Implementation.DebuggedProcess.<StartDebugAdapter>b__111_3(Exception ex)
Microsoft.VisualStudio.Debugger.VSCodeDebuggerHost.Utilities.TaskExtensions.<>c__DisplayClass11_0`1.<Catch>b__0(TException ex)
Microsoft.VisualStudio.Debugger.VSCodeDebuggerHost.Utilities.TaskExtensions.<>c__DisplayClass10_0`1.<Catch>b__0(AggregateException ex)

    Failure Location: LaunchRequest
    Inner Exception: 
        ProtocolException: launch: program '' does not exist.

 1> ERROR: One or more errors occurred.

Failed to launch debug adapter.  Additional information may be available in the output window.

launch: program '' does not exist.
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"response\",\"request_seq\":2,\"success\":true,\"command\":\"handshake\",\"body\":{\"signature\":\"93959a+gFuOZFsItzAQMPqUeD2aiS6LBd+ZFn5DARi2II8=\"},\"seq\":3}\n"},"seq":9}
 1> WARNING: Ignoring event of type 'output' received after moving to 'Exited' state!
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":10,\"type\":\"event\",\"event\":\"initialized\",\"body\":{}}\n"},"seq":11}
 1> WARNING: Ignoring event of type 'output' received after moving to 'Exited' state!
 1> [DebugAdapter] <--   E (initialized): {"type":"event","event":"initialized","body":{},"seq":10}
 1> WARNING: Ignoring event of type 'initialized' received after moving to 'Exited' state!
 1> Debug adapter process exited.
```

#### dotnet --info on device:
```
  It was not possible to find any installed .NET Core SDKs
  Did you mean to run .NET Core SDK commands? Install a .NET Core SDK from:
      https://aka.ms/dotnet-download

Host (useful for support):
  Version: 3.1.11
  Commit:  f5eceb8105

.NET Core SDKs installed:
  No SDKs were found.

.NET Core runtimes installed:
  Microsoft.NETCore.App 3.1.11 [/usr/share/dotnet31/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by hoyosjs** [comment]
> * Why should I clone coreclr from github? I've downloaded v3.1.11 from here: https://dotnet.microsoft.com/download/dotnet-core/3.1 and installed it on the device. Is that just needed to look at the source code or do I miss something?

Yes, easier to follow osurce to see where to step in/over.

> * I'm not using public key auth on ssh, just entering the username (root) is enough to login (Device is inside local network only). Is that a problem?

Not at all.

> * I've made some changes to your launch.json script (left out the --pauseEngineForDebugger for now)
 
It's ok to do this just to test setup.

> Unfortunately I get an error when launching `DebugAdapterHost.Launch /LaunchJson:launch.json`:

I glanced over this. Not sure, but it sounds like you need to change program to a full path: `"program": "/usr/share/dotnet/dotnet"`

**Comment by XXXBold** [comment]
> 
> 
> I glanced over this. Not sure, but it sounds like you need to change program to a full path: `"program": "/usr/share/dotnet/dotnet"`

Yes, that fixed it, thanks.

launch.json now looks like this:
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "$adapter": "plink.exe",
      "$adapterArgs": "root@10.10.11.55 -batch -T /home/root/vsdbg/vsdbg --pauseEngineForDebugger --interpreter=vscode",
      "name": "Launch REMOTE",
      "type": "coreclr",
      "request": "launch",
      "program": "/usr/share/dotnet31/dotnet",
      "args": [ "/home/root/dn31dbg/ConsoleApp1.dll" ],
      "cwd": "/home/root/dn31dbg/",
      "stopAtEntry": true,
      "console": "internalConsole",
      "justMyCode": false,
      "logging": {
        "engineLogging": true,
        "programOutput": true,
        "exceptions": true
      }
    }
  ]
}
```

I was able to run it now, and attach the debugger to the process (PID 893):
![htop-31-dbg](https://user-images.githubusercontent.com/30773753/107616080-9f8c8480-6c4d-11eb-81a9-242ba76e33eb.PNG)


Not sure I've made this correctly, didn't copy over the source code, so the .cpp files were not found while debugging. Still, hope it's usefull for you, if not, let me know what to do otherwise. 

Here's GDBs output from the device:

```
root@colibri-t30:~# gdb
GNU gdb (GDB) 8.0
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-angstrom-linux-gnueabi".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
(gdb) attach 893
Attaching to process 893
[New LWP 894]
warning: File "/lib/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/aut                                                           o-load".
To enable execution of this file add
        add-auto-load-safe-path /lib/libthread_db-1.0.so
line to your configuration file "/home/root/.gdbinit".
To completely disable this security protection add
        set auto-load safe-path /
line to your configuration file "/home/root/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
        info "(gdb)Auto-loading safe path"
warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
warning: File "/lib/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/aut                                                           o-load".
warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
0x40684e4c in pthread_cond_timedwait () from /lib/libpthread.so.0
(gdb) break ClrDataAccess::GetDacGlobals
Function "ClrDataAccess::GetDacGlobals" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (ClrDataAccess::GetDacGlobals) pending.
(gdb) continue
Continuing.
[New LWP 899]
[New LWP 900]
[New LWP 901]
[New LWP 902]
[New LWP 903]
[New LWP 904]
[New LWP 905]
[New LWP 906]
[New LWP 907]
[New LWP 908]
[New LWP 909]
[New LWP 911]
[New LWP 912]
[New LWP 913]
[New LWP 917]
[New LWP 919]
[LWP 907 exited]
[New LWP 918]
[New LWP 920]
[New LWP 921]
[Switching to LWP 920]

Thread 20 "vsdbg" hit Breakpoint 1, ClrDataAccess::Initialize (this=0x954190) at /__w/1/s/src/debug/daccess/daccess.cpp:5595
5595    /__w/1/s/src/debug/daccess/daccess.cpp: No such file or directory.
(gdb) step
ClrDataAccess::GetDacGlobals (this=<optimized out>) at /__w/1/s/src/debug/daccess/daccess.cpp:7260
7260    in /__w/1/s/src/debug/daccess/daccess.cpp
(gdb) step
[LWP 917 exited]
[LWP 906 exited]
7259    in /__w/1/s/src/debug/daccess/daccess.cpp
(gdb) step
7260    in /__w/1/s/src/debug/daccess/daccess.cpp
(gdb) step
[LWP 918 exited]
ReadFromDataTarget (pDataTarget=0x94d0a0, addr=<optimized out>, pBuffer=0x4e3aad60 <g_dacGlobals> "", bytesToRead=<optimized out>)
    at /__w/1/s/src/debug/dbgutil/dbgutil.cpp:405
405     /__w/1/s/src/debug/dbgutil/dbgutil.cpp: No such file or directory.
(gdb) step
408     in /__w/1/s/src/debug/dbgutil/dbgutil.cpp
(gdb) step
ShimRemoteDataTarget::ReadVirtual (this=0x94d0a0, address=1083790916,
    pBuffer=0x4e3a0738 <vtable for DacDbiInterfaceImpl+1240> "e\334\034N\237\334\034N\243\335\034N;W\036N\370\377\377\377\300\v:Ni\334\034                                                     N\275\334\034N\331\335\034N-\031\037N\005\203\037N\001\201\037N\035\206\037N\241\222\037N\361F\037Na\216\037N]\225\037Nɘ\037NaU\037N\021Y\                                                     037N\215S\037N\225\235\037N\331Z\037N}1\037Ne*\037N\311$\037N96\037N\371?\037NeE\037N\021>\037N\355O\037N\241\071\037N5<\037Ne\035\037N\02                                                     1\337\037N=\337\037Ny\026\037N\255\020\037N\201\022\037N\345~\037N\275I\037N\265M\037Nuc\037N\361]\037N\335\037\037N\021l\037N\351o\037N\3                                                     15q\037N\035i\037N\375s\037N"..., cbRequestSize=0, pcbRead=0x94d0a0)
    at /__w/1/s/src/debug/di/shimremotedatatarget.cpp:267
267     /__w/1/s/src/debug/di/shimremotedatatarget.cpp: No such file or directory.
(gdb) step
273     in /__w/1/s/src/debug/di/shimremotedatatarget.cpp
(gdb) step
275     in /__w/1/s/src/debug/di/shimremotedatatarget.cpp
(gdb) step
PAL__pread () at /__w/1/s/bin/obj/Linux.arm.Release/src/inc/palredefines.inc:182
182     /__w/1/s/bin/obj/Linux.arm.Release/src/inc/palredefines.inc: No such file or directory.
(gdb) step
DAC_PAL__pread () at /__w/1/s/bin/obj/Linux.arm.Release/src/inc/libredefines.inc:182
182     /__w/1/s/bin/obj/Linux.arm.Release/src/inc/libredefines.inc: No such file or directory.
(gdb) step
PAL__pread (fd=58, buf=0x4e3aad60 <g_dacGlobals>, nbytes=712, offset=1083790916) at /__w/1/s/src/pal/src/cruntime/file.cpp:466
466     /__w/1/s/src/pal/src/cruntime/file.cpp: No such file or directory.
(gdb) step
0x40689568 in pread64 () from /lib/libpthread.so.0
(gdb) step
Single stepping until exit from function pread64,
which has no line number information.
Warning:
Cannot insert breakpoint 0.
Cannot access memory at address 0x3a

0x40688228 in ?? () from /lib/libpthread.so.0
(gdb) step
Cannot find bounds of current function
(gdb) step
Cannot find bounds of current function
(gdb) backtrace
#0  0x40688228 in ?? () from /lib/libpthread.so.0
#1  0x0000003a in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) continue
Continuing.
[LWP 920 exited]
[LWP 912 exited]
```

After the continue at the end, the usual error pops up.

And here's the Debug Adapter Host Log from within VS:
```
1> DebugAdapterHost version: 16.8.40922.1 commit:6ac6030cbe2c515a73b952feec2d243d418d0915
 1> Starting 'plink.exe' with arguments 'root@10.10.11.55 -batch -T /home/root/vsdbg/vsdbg --pauseEngineForDebugger --interpreter=vscode'
 1> WARNING: 'Language' not specified in engine registration!
 1> WARNING: 'LanguageId' not specified in engine registration!
 1> [DebugAdapter] --> C (initialize-1): {"type":"request","command":"initialize","arguments":{"pathFormat":"path","clientID":"visualstudio","clientName":"Visual Studio","adapterID":"coreclr","locale":"en-US","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsRunInTerminalRequest":true,"supportsMemoryReferences":true,"supportsProgressReporting":true,"SupportsMessageBox":true,"supportsHandshakeRequest":true,"supportsVsAdditionalBreakpointBinds":true,"supportsHitCountsChange":true,"supportsVsCustomMessages":true},"seq":1}
 1> WARNING: Request 'initialize-1' has not received a response within 1000 ms!
 1> [DebugAdapter] <--   R (initialize-1) [88887 ms]: {"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsHitConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"exceptionBreakpointFilters":[{"filter":"all","label":"All Exceptions","default":false},{"filter":"user-unhandled","label":"User-Unhandled Exceptions","default":true}],"supportsSetVariable":true,"supportsGotoTargetsRequest":true,"supportsModulesRequest":true,"additionalModuleColumns":[{"attributeName":"vsLoadAddress","label":"Load Address","type":"string"},{"attributeName":"vsPreferredLoadAddress","label":"Preferred Load Address","type":"string"},{"attributeName":"vsModuleSize","label":"Module Size","type":"number"},{"attributeName":"vsLoadOrder","label":"Order","type":"number"},{"attributeName":"vsTimestampUTC","label":"Timestamp","type":"unixTimestampUTC"},{"attributeName":"vsIs64Bit","label":"64-bit","type":"boolean"},{"attributeName":"vsAppDomain","label":"AppDomain","type":"string"},{"attributeName":"vsAppDomainId","label":"AppDomainId","type":"number"}],"supportedChecksumAlgorithms":["MD5","SHA1","SHA256"],"supportsExceptionOptions":true,"supportsValueFormattingOptions":true,"supportsExceptionInfoRequest":true,"supportTerminateDebuggee":true,"supportsSetExpression":true,"supportsReadMemoryRequest":true,"supportsCancelRequest":true,"supportsExceptionConditions":true,"supportsLoadSymbolsRequest":true,"supportsModuleSymbolSearchLog":true,"supportsDebuggerProperties":true,"supportsSetSymbolOptions":true,"supportsHitBreakpointIds":true,"supportsVsIndividualBreakpointOperations":true,"supportsVsBreakpointLanguage":true,"supportsSetHitCount":true,"supportsVsCustomMessages":true,"supportsEvaluationOptions":true,"supportsExceptionStackTrace":true,"memoryReferencesAreAddresses":true,"supportsObjectFavorites":true,"supportsObjectId":true,"supportsVariableEnumerators":false},"seq":1}
 1> [DebugAdapter] --> C (launch-2): {"type":"request","command":"launch","arguments":{"name":"Launch REMOTE","type":"coreclr","request":"launch","program":"/usr/share/dotnet31/dotnet","args":["/home/root/dn31dbg/ConsoleApp1.dll"],"cwd":"/home/root/dn31dbg/","stopAtEntry":true,"console":"internalConsole","justMyCode":false,"logging":{"engineLogging":true,"programOutput":true,"exceptions":true}},"seq":2}
 1> [DebugAdapter] <--   C (handshake-2): {"type":"request","command":"handshake","arguments":{"value":"nTpP//kbNm1+IhUXk5wJbhhbU8+q8X72AfkfD6WOj2g="},"seq":2}
 1> [DebugAdapter] --> R (handshake-2): {"type":"response","request_seq":2,"success":true,"command":"handshake","body":{"signature":"685w4kJ0htwauKNXzQaw35EVJtTutY4yscXhuswyJY32BA="},"seq":3}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":3,\"type\":\"event\",\"event\":\"output\",\"body\":{\"category\":\"console\",\"output\":\"-------------------------------------------------------------------\\nYou may only use the Microsoft .NET Core Debugger (vsdbg) with\\nVisual Studio Code, Visual Studio or Visual Studio for Mac software\\nto help you develop and test your applications.\\n-------------------------------------------------------------------\\n\",\"severity\":\"ok\"}}\n"},"seq":4}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-------------------------------------------------------------------\nYou may only use the Microsoft .NET Core Debugger (vsdbg) with\nVisual Studio Code, Visual Studio or Visual Studio for Mac software\nto help you develop and test your applications.\n-------------------------------------------------------------------\n","severity":"ok"},"seq":3}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":5,\"type\":\"response\",\"request_seq\":2,\"success\":true,\"command\":\"launch\"}\n"},"seq":6}
 1> [DebugAdapter] <--   R (launch-2) [71 ms]: {"type":"response","request_seq":2,"success":true,"command":"launch","seq":5}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"response\",\"request_seq\":2,\"success\":true,\"command\":\"handshake\",\"body\":{\"signature\":\"685w4kJ0htwauKNXzQaw35EVJtTutY4yscXhuswyJY32BA=\"},\"seq\":3}\n"},"seq":7}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":8,\"type\":\"event\",\"event\":\"initialized\",\"body\":{}}\n"},"seq":9}
 1> [DebugAdapter] <--   E (initialized): {"type":"event","event":"initialized","body":{},"seq":8}
 1> [DebugAdapter] --> C (setDebuggerProperty-4): {"type":"request","command":"setDebuggerProperty","arguments":{"AllowOutOfProcessSymbols":1},"seq":4}
 1> [DebugAdapter] --> C (setDebuggerProperty-5): {"type":"request","command":"setDebuggerProperty","arguments":{"DisableJITOptimization":0},"seq":5}
 1> [DebugAdapter] --> C (setDebuggerProperty-6): {"type":"request","command":"setDebuggerProperty","arguments":{"EnableFastEvaluate":1},"seq":6}
 1> [DebugAdapter] --> C (setDebuggerProperty-7): {"type":"request","command":"setDebuggerProperty","arguments":{"InterpreterOptions":1},"seq":7}
 1> [DebugAdapter] --> C (setDebuggerProperty-8): {"type":"request","command":"setDebuggerProperty","arguments":{"JustMyCodeStepping":1},"seq":8}
 1> [DebugAdapter] --> C (setDebuggerProperty-9): {"type":"request","command":"setDebuggerProperty","arguments":{"StopOnExceptionCrossingManagedBoundary":0},"seq":9}
 1> [DebugAdapter] --> C (setDebuggerProperty-10): {"type":"request","command":"setDebuggerProperty","arguments":{"WarnIfNoUserCodeOnLaunch":1},"seq":10}
 1> [DebugAdapter] --> C (setDebuggerProperty-11): {"type":"request","command":"setDebuggerProperty","arguments":{"EnableStepFiltering":true},"seq":11}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"AllowOutOfProcessSymbols\":1},\"seq\":4}\n"},"seq":10}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":11,\"type\":\"response\",\"request_seq\":4,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":12}
 1> [DebugAdapter] <--   R (setDebuggerProperty-4) [9 ms]: {"type":"response","request_seq":4,"success":true,"command":"setDebuggerProperty","body":{},"seq":11}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"DisableJITOptimization\":0},\"seq\":5}\n"},"seq":13}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":14,\"type\":\"response\",\"request_seq\":5,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":15}
 1> [DebugAdapter] --> C (setSymbolOptions-12): {"type":"request","command":"setSymbolOptions","arguments":{"symbolOptions":{"cachePath":"","moduleFilter":{"mode":"loadAllButExcluded","excludedModules":[""]}}},"seq":12}
 1> [DebugAdapter] <--   R (setDebuggerProperty-5) [25 ms]: {"type":"response","request_seq":5,"success":true,"command":"setDebuggerProperty","body":{},"seq":14}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"EnableFastEvaluate\":1},\"seq\":6}\n"},"seq":16}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":17,\"type\":\"response\",\"request_seq\":6,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":18}
 1> [DebugAdapter] <--   R (setDebuggerProperty-6) [26 ms]: {"type":"response","request_seq":6,"success":true,"command":"setDebuggerProperty","body":{},"seq":17}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"InterpreterOptions\":1},\"seq\":7}\n"},"seq":19}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":20,\"type\":\"response\",\"request_seq\":7,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":21}
 1> [DebugAdapter] <--   R (setDebuggerProperty-7) [27 ms]: {"type":"response","request_seq":7,"success":true,"command":"setDebuggerProperty","body":{},"seq":20}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"JustMyCodeStepping\":1},\"seq\":8}\n"},"seq":22}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":23,\"type\":\"response\",\"request_seq\":8,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":24}
 1> [DebugAdapter] <--   R (setDebuggerProperty-8) [28 ms]: {"type":"response","request_seq":8,"success":true,"command":"setDebuggerProperty","body":{},"seq":23}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"StopOnExceptionCrossingManagedBoundary\":0},\"seq\":9}\n"},"seq":25}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":26,\"type\":\"response\",\"request_seq\":9,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":27}
 1> [DebugAdapter] <--   R (setDebuggerProperty-9) [28 ms]: {"type":"response","request_seq":9,"success":true,"command":"setDebuggerProperty","body":{},"seq":26}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"WarnIfNoUserCodeOnLaunch\":1},\"seq\":10}\n"},"seq":28}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":29,\"type\":\"response\",\"request_seq\":10,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":30}
 1> [DebugAdapter] <--   R (setDebuggerProperty-10) [29 ms]: {"type":"response","request_seq":10,"success":true,"command":"setDebuggerProperty","body":{},"seq":29}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setDebuggerProperty\",\"arguments\":{\"EnableStepFiltering\":true},\"seq\":11}\n"},"seq":31}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":32,\"type\":\"response\",\"request_seq\":11,\"success\":true,\"command\":\"setDebuggerProperty\",\"body\":{}}\n"},"seq":33}
 1> [DebugAdapter] <--   R (setDebuggerProperty-11) [30 ms]: {"type":"response","request_seq":11,"success":true,"command":"setDebuggerProperty","body":{},"seq":32}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setSymbolOptions\",\"arguments\":{\"symbolOptions\":{\"cachePath\":\"\",\"moduleFilter\":{\"mode\":\"loadAllButExcluded\",\"excludedModules\":[\"\"]}}},\"seq\":12}\n"},"seq":34}
 1> [DebugAdapter] --> C (setFunctionBreakpoints-13): {"type":"request","command":"setFunctionBreakpoints","arguments":{"breakpoints":[]},"seq":13}
 1> [DebugAdapter] --> C (setBreakpoints-14): {"type":"request","command":"setBreakpoints","arguments":{"source":{"path":"c:\\Projects\\VSProjects\\ConsoleApp1\\Program.vb","sources":[],"checksums":[{"algorithm":"MD5","checksum":"69d4e2cad9f070336ebbb0637d59a7da"},{"algorithm":"MD5","checksum":"d802c1320cabfda4c882d1f6caa4d91b"},{"algorithm":"SHA1","checksum":"406232dbf3fbb62fc40c6deca0a70953c00840bf"},{"algorithm":"SHA1","checksum":"ea5ba904b5e9bb4a5c35977639ccb6ab4e7d34d1"},{"algorithm":"SHA256","checksum":"19bd9a797dcaac14c1ab798ed08cc6f3a9d67894184bdfc79265dd63fb9f64a5"},{"algorithm":"SHA256","checksum":"2ac10c7e7a0b4880278d7af4c13b805c1bc37b3b59365999efafa5f3c3615dbe"}]},"breakpoints":[{"line":6,"column":13}],"lines":[6]},"seq":14}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":35,\"type\":\"response\",\"request_seq\":12,\"success\":true,\"command\":\"setSymbolOptions\"}\n"},"seq":36}
 1> [DebugAdapter] <--   R (setSymbolOptions-12) [643 ms]: {"type":"response","request_seq":12,"success":true,"command":"setSymbolOptions","seq":35}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setFunctionBreakpoints\",\"arguments\":{\"breakpoints\":[]},\"seq\":13}\n"},"seq":37}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":38,\"type\":\"response\",\"request_seq\":13,\"success\":true,\"command\":\"setFunctionBreakpoints\",\"body\":{\"breakpoints\":[]}}\n"},"seq":39}
 1> [DebugAdapter] <--   R (setFunctionBreakpoints-13) [622 ms]: {"type":"response","request_seq":13,"success":true,"command":"setFunctionBreakpoints","body":{"breakpoints":[]},"seq":38}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setBreakpoints\",\"arguments\":{\"source\":{\"path\":\"c:\\\\Projects\\\\VSProjects\\\\ConsoleApp1\\\\Program.vb\",\"sources\":[],\"checksums\":[{\"algorithm\":\"MD5\",\"checksum\":\"69d4e2cad9f070336ebbb0637d59a7da\"},{\"algorithm\":\"MD5\",\"checksum\":\"d802c1320cabfda4c882d1f6caa4d91b\"},{\"algorithm\":\"SHA1\",\"checksum\":\"406232dbf3fbb62fc40c6deca0a70953c00840bf\"},{\"algorithm\":\"SHA1\",\"checksum\":\"ea5ba904b5e9bb4a5c35977639ccb6ab4e7d34d1\"},{\"algorithm\":\"SHA256\",\"checksum\":\"19bd9a797dcaac14c1ab798ed08cc6f3a9d67894184bdfc79265dd63fb9f64a5\"},{\"algorithm\":\"SHA256\",\"checksum\":\"2ac10c7e7a0b4880278d7af4c13b805c1bc37b3b59365999efafa5f3c3615dbe\"}]},\"breakpoints\":[{\"line\":6,\"column\":13}],\"lines\":[6]},\"seq\":14}\n"},"seq":40}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":41,\"type\":\"response\",\"request_seq\":14,\"success\":true,\"command\":\"setBreakpoints\",\"body\":{\"breakpoints\":[{\"id\":1,\"verified\":false,\"message\":\"The breakpoint is pending and will be resolved when debugging starts.\"}]}}\n"},"seq":42}
 1> [DebugAdapter] <--   R (setBreakpoints-14) [625 ms]: {"type":"response","request_seq":14,"success":true,"command":"setBreakpoints","body":{"breakpoints":[{"id":1,"verified":false,"message":"The breakpoint is pending and will be resolved when debugging starts."}]},"seq":41}
 1> [DebugAdapter] --> C (setExceptionBreakpoints-15): {"type":"request","command":"setExceptionBreakpoints","arguments":{"filters":[],"exceptionOptions":[]},"seq":15}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"setExceptionBreakpoints\",\"arguments\":{\"filters\":[],\"exceptionOptions\":[]},\"seq\":15}\n"},"seq":43}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":44,\"type\":\"response\",\"request_seq\":15,\"success\":true,\"command\":\"setExceptionBreakpoints\"}\n"},"seq":45}
 1> [DebugAdapter] <--   R (setExceptionBreakpoints-15) [16 ms]: {"type":"response","request_seq":15,"success":true,"command":"setExceptionBreakpoints","seq":44}
 1> [DebugAdapter] --> C (configurationDone-16): {"type":"request","command":"configurationDone","arguments":{},"seq":16}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"-> (C) {\"type\":\"request\",\"command\":\"configurationDone\",\"arguments\":{},\"seq\":16}\n"},"seq":46}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":47,\"type\":\"event\",\"event\":\"output\",\"body\":{\"category\":\"telemetry\",\"output\":\"VS/Diagnostics/Debugger/vsdbg/ProcessCreate\",\"data\":{\"VS.Diagnostics.Debugger.vsdbg.Distribution.Version\":\"\",\"VS.Diagnostics.Debugger.vsdbg.Distribution.Name\":\"e05d318\",\"VS.Diagnostics.Debugger.vsdbg.OSFamily\":\"Linux\",\"VS.Diagnostics.Debugger.vsdbg.Version\":\"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3\"}}}\n"},"seq":48}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/ProcessCreate","data":{"VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"e05d318","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.Version":"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3"}},"seq":47}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":49,\"type\":\"event\",\"event\":\"telemetryDetails\",\"body\":{\"telemetryEventName\":\"vs/diagnostics/debugger/process/context\",\"telemetryProperties\":{\"VS.Diagnostics.Debugger.AD7Process.Architecture\":\"ARM\"}}}\n"},"seq":50}
 1> [DebugAdapter] <--   E (telemetryDetails): {"type":"event","event":"telemetryDetails","body":{"telemetryEventName":"vs/diagnostics/debugger/process/context","telemetryProperties":{"VS.Diagnostics.Debugger.AD7Process.Architecture":"ARM"}},"seq":49}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":51,\"type\":\"event\",\"event\":\"process\",\"body\":{\"name\":\"/usr/share/dotnet31/dotnet\",\"systemProcessId\":910,\"startMethod\":\"launch\",\"pointerSize\":32}}\n"},"seq":52}
 1> [DebugAdapter] <--   E (process): {"type":"event","event":"process","body":{"name":"/usr/share/dotnet31/dotnet","systemProcessId":910,"startMethod":"launch","pointerSize":32},"seq":51}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":53,\"type\":\"event\",\"event\":\"breakpoint\",\"body\":{\"reason\":\"changed\",\"breakpoint\":{\"id\":1,\"verified\":false,\"message\":\"No symbols have been loaded for this document.\"}}}\n"},"seq":54}
 1> [DebugAdapter] <--   E (breakpoint): {"type":"event","event":"breakpoint","body":{"reason":"changed","breakpoint":{"id":1,"verified":false,"message":"No symbols have been loaded for this document."}},"seq":53}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":55,\"type\":\"event\",\"event\":\"output\",\"body\":{\"category\":\"telemetry\",\"output\":\"VS/Diagnostics/Debugger/vsdbg/Launch\",\"data\":{\"VS.Diagnostics.Debugger.vsdbg.AdapterId\":\"coreclr\",\"VS.Diagnostics.Debugger.vsdbg.Launch.Duration\":1140,\"VS.Diagnostics.Debugger.vsdbg.VisualizerFileUsed\":false,\"VS.Diagnostics.Debugger.vsdbg.TargetType\":\"Live\",\"VS.Diagnostics.Debugger.vsdbg.Distribution.Version\":\"\",\"VS.Diagnostics.Debugger.vsdbg.Distribution.Name\":\"e05d318\",\"VS.Diagnostics.Debugger.vsdbg.OSFamily\":\"Linux\",\"VS.Diagnostics.Debugger.vsdbg.SourceFileMappings\":0,\"VS.Diagnostics.Debugger.vsdbg.Version\":\"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3\"}}}\n"},"seq":56}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/Launch","data":{"VS.Diagnostics.Debugger.vsdbg.AdapterId":"coreclr","VS.Diagnostics.Debugger.vsdbg.Launch.Duration":1140,"VS.Diagnostics.Debugger.vsdbg.VisualizerFileUsed":false,"VS.Diagnostics.Debugger.vsdbg.TargetType":"Live","VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"e05d318","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.SourceFileMappings":0,"VS.Diagnostics.Debugger.vsdbg.Version":"16.9.20122.2 commit:c3006f59a1d3a160c9fc06f5e2e14402522190e3"}},"seq":55}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (R) {\"seq\":57,\"type\":\"response\",\"request_seq\":16,\"success\":true,\"command\":\"configurationDone\"}\n"},"seq":58}
 1> [DebugAdapter] <--   R (configurationDone-16) [358 ms]: {"type":"response","request_seq":16,"success":true,"command":"configurationDone","seq":57}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"console","output":"<- (E) {\"seq\":59,\"type\":\"event\",\"event\":\"output\",\"body\":{\"category\":\"messageBox\",\"output\":\"Unable to attach to CoreCLR. Unknown Error: 0x80131c4f\",\"severity\":\"error\"}}\n"},"seq":60}
 1> [DebugAdapter] <--   E (output): {"type":"event","event":"output","body":{"category":"messageBox","output":"Unable to attach to CoreCLR. Unknown Error: 0x80131c4f","severity":"error"},"seq":59}

```

**Comment by XXXBold** [comment]
Any updates on this?

**Comment by hoyosjs** [comment]
Sorry it took so long to respond to this - it fell through the cracks of the last release. There were several fixes in this space with respect to ARM32, sign extended syscalls, and debugging APIs. If this is still failing, feel free to ping again and we can get some more information. This issue is not very actionable in its current state, and we haven't gotten any further reports of this.
